client/src/components/DataEntry/Ballot.tsx

```diff
@@ -77,12 +143,49 @@ const Ballot: React.FC<IProps> = ({
   const [interpretations, setInterpretations] = useState<
     IBallotInterpretation[]
   >(contests.map(emptyInterpretation))
+  const { confirm, confirmProps } = useConfirm()
+
+  const setInterpretationsFunc = (
+    newInterpretations: IBallotInterpretation[]
+  ) => {
+    setInterpretations(newInterpretations)
+    setAuditing(false)
+  }
 
   const submitNotFound = async () => {
     await submitBallot(ballot.id, BallotStatus.NOT_FOUND, [])
     nextBallot()
   }
 
+  useEffect(() => {
+    if (!auditing) {
+      confirm({
+        interpretations,
+        contests,
+        onYesClick: async () => {
+          submitBallot(
+            ballot.id,
+            BallotStatus.AUDITED,
+            interpretations.filter(
+              ({ interpretation }) => interpretation !== null
+            )
+          )
+          setAuditing(true)
+          nextBallot()
+        },
+      })
+      setAuditing(true)
+    }
+  }, [
+    auditing,
+    confirm,
+    interpretations,
+    contests,
+    submitBallot,
+    ballot,
+    nextBallot,
+  ])
```

Comment:
You should be able to just call `confirm` directly when the user clicks the "Submit Selections" button instead of using this intermediate `auditing` state. Any particular reason for the `auditing` state?